pkg/spflib/flatten_test.go

```diff
@@ -107,17 +109,91 @@ func TestSplit(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
-			res := rec.TXTSplit("_spf%d.stackex.com", overhead)
-			if res["@"] != tst[2] {
-				t.Fatalf("Root record wrong. \nExp %s\ngot %s", tst[2], res["@"])
+			res := rec.TXTSplit("_spf%d.stackex.com", overhead, txtMaxSize)
```

Comment:
Dumb question but...

I see that on my test domain the result was a TXT record like:

> _spf1      300   IN TXT   "v=spf1 ip4:209.85.128.0/17 ip4:216.58.192.0/19 ip4:216.239.32.0/19 ip6:2001:4860:4000::/36 ip6:2404:6800:4000::/36 ip6:2607:f8b0:4000::/36 ip6:2800:3f0:4000::/36 ip6:2a00:1450:4000::/36 ip6:2c0f:fb50:4000::/36 ip4:172.217.0.0/19 ip4:172.217.32.0/20 ip4:17" "2.217.128.0/19 ip4:172.217.160.0/20 ip4:172.217.192.0/19 ip4:172.253.56.0/21 ip4:172.253.112.0/20 ip4:108.177.96.0/19 ip4:35.191.0.0/16 ip4:130.211.0.0/22 include:_spf2.stackoverflow.com ~all"

I haven't read the RFCs but is splitting the string anywhere permitted or does it have to be on a space boundary?   i.e. I notice `ip4:17" "2.217.128.0/19` and that makes me wonder if it the split is permitted there.

There are other RFCs where the individual strings are joined with a space.: `strings.Join(list, "")` vs `strings.Join(list, " ")`

Reply:
Not a dumb question at all. It caught me off guard as well since it just looks wrong. 

According to the RFC you can split anywhere and you don't add any extra white space. 

From: https://tools.ietf.org/html/rfc4408#section-3.1.3

> As defined in [RFC1035] sections 3.3.14 and 3.3, a single text DNS
> record (either TXT or SPF RR types) can be composed of more than one
> string.  If a published record contains multiple strings, then the
> record MUST be treated as if those strings are concatenated together
> without adding spaces.  For example:
> 
>    IN TXT "v=spf1 .... first" "second string..."
> 
> MUST be treated as equivalent to
> 
>    IN TXT "v=spf1 .... firstsecond string..."
> 
> SPF or TXT records containing multiple strings are useful in
> constructing records that would exceed the 255-byte maximum length of
> a string within a single TXT or SPF RR record.

I have been testing these changes with multiple strings against https://www.kitterman.com/spf/validate.html and a few other tools to double-check that they are valid.

If you want to check for yourself stspg-customer.com is an example of a provider that is using multiple strings that passes in kitterman.

```
❯ dig txt stspg-customer.com +short
"v=spf1 ip4:23.253.182.103 ip4:23.253.183.145 ip4:23.253.183.146 ip4:23.253.183.147 ip4:23.253.183.148 ip4:23.253.183.150 ip4:166.78.68.221 ip4:166.78.69.146 ip4:167.89.46.159 ip4:167.89.64.9 ip4:167.89.65.0 ip4:167.89.65.53 ip4:167.89.65.100 ip4:167.89.74" ".233 ip4:167.89.75.33 ip4:167.89.75.126 ip4:167.89.75.136 ip4:167.89.75.164 ip4:192.237.159.42 ip4:192.237.159.43 -all"
```

Reply:
Great!  Thanks for the explanation (translation: thank you for not making me read the RFC :-) )